Serving images (only) do not seem to have attestation information.

Works:

```
cosign download attestation $(curl -L https://github.com/knative/eventing/releases/download/knative-v1.12.1/eventing-core.yaml | grep 'image: [a-z]' | head -1 | cut -d: -f2-) | jq -r .payload | base64 --decode | jq
```

Does not work:

```
cosign download attestation $(curl -L https://github.com/knative/serving/releases/download/knative-v1.12.2/serving-core.yaml | grep 'image: [a-z]' | head -1 | cut -d: -f2-) | jq -r .payload | base64 --decode | jq
```

(Error is: `main.go:74: error during command execution: found no attestations`). I think this is because we aren't setting `ATTEST_IMAGES` in the serving release job. A `grep -L ATTEST_IMAGES *.yaml` suggests that Serving is the only repo missed.

I have a PR.

--- a/prow/jobs/generated/knative/serving-main.gen.yaml
+++ b/prow/jobs/generated/knative/serving-main.gen.yaml
@@ -774,6 +774,10 @@ periodics:
         value: us-central1
       - name: GOOGLE_APPLICATION_CREDENTIALS
         value: /etc/nightly-account/service-account.json
+      - name: SIGN_IMAGES
+        value: "true"
+      - name: ATTEST_IMAGES
+        value: "true"
       image: us-docker.pkg.dev/knative-tests/images/prow-tests:v20231207-30ee442a2
       name: ""
       resources:
@@ -846,6 +850,10 @@ periodics:
         value: "true"
       - name: E2E_CLUSTER_REGION
         value: us-central1
+      - name: SIGN_IMAGES
+        value: "true"
+      - name: ATTEST_IMAGES
+        value: "true"
       image: us-docker.pkg.dev/knative-tests/images/prow-tests:v20231207-30ee442a2
       name: ""
       resources:

--- a/prow/jobs/generated/knative/serving-main.gen.yaml
+++ b/prow/jobs/generated/knative/serving-main.gen.yaml
@@ -855,6 +855,10 @@ periodics:
         value: us-central1
       - name: GOOGLE_APPLICATION_CREDENTIALS
         value: /etc/nightly-account/service-account.json
+      - name: SIGN_IMAGES
+        value: "true"
+      - name: ATTEST_IMAGES
+        value: "true"
       image: us-docker.pkg.dev/knative-tests/images/prow-tests:v20231129-81431e1fe
       name: ""
       resources:
@@ -927,6 +931,10 @@ periodics:
         value: "true"
       - name: E2E_CLUSTER_REGION
         value: us-central1
+      - name: SIGN_IMAGES
+        value: "true"
+      - name: ATTEST_IMAGES
+        value: "true"
       image: us-docker.pkg.dev/knative-tests/images/prow-tests:v20231129-81431e1fe
       name: ""
       resources:

--- a/prow/jobs_config/knative/serving.yaml
+++ b/prow/jobs_config/knative/serving.yaml
@@ -283,6 +283,10 @@ requirement_presets:
         value: us-central1
       - name: GOOGLE_APPLICATION_CREDENTIALS
         value: /etc/nightly-account/service-account.json
+      - name: SIGN_IMAGES
+        value: "true"
+      - name: ATTEST_IMAGES
+        value: "true"
     volumeMounts:
       - name: nightly-account
         mountPath: /etc/nightly-account
@@ -301,6 +305,10 @@ requirement_presets:
     env:
       - name: E2E_CLUSTER_REGION
         value: us-central1
+      - name: SIGN_IMAGES
+        value: "true"
+      - name: ATTEST_IMAGES
+        value: "true"
     volumeMounts:
       - name: hub-token
         mountPath: /etc/hub-token

